chans/chans.go

```diff
@@ -41,16 +74,33 @@ func ForEach[T any](source <-chan T, f func(T)) {
 	}()
 }
 
-// Flatten flattens a chan of chans into a chan of elements
-func Flatten[T any](source <-chan <-chan T) <-chan T {
-	out := make(chan T, cap(source))
+// FromValues creates a ComposableChannel from a list of values
+func FromValues[T any](values ...T) <-chan T {
+	newCh := make(chan T, len(values))
+	defer close(newCh)
+
+	for _, v := range values {
+		newCh <- v
+	}
+
+	return newCh
+}
+
+// Map maps a channel of T to a channel of S. Runs until source channel is closed.
+// By default, output channel has the same capacity as the source channel.
+// Desired capacity of the output channel can be specified via an optional argument.
+func Map[T, S any](source <-chan T, f func(T) S, capacity ...int) <-chan S {
+	var out chan S
+	if len(capacity) > 0 {
+		out = make(chan S, capacity[0])
+	} else {
+		out = make(chan S, cap(source))
```

Comment:
```suggestion
		out = make(chan S)
```
if we use `cap(source)` we effectively double the capacity of the channel

Reply:
I prefer unbuffered too, just kept it the same as it was before the change. 👍